### PR TITLE
[feature-wip] (arrow-flight) (step1) BE support Arrow Flight server, read data only

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Design Documentation Linked to https://github.com/apache/doris/issues/25514

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/common/config.cpp [hunk 1]
```diff
@@ -60,6 +60,9 @@ DEFINE_Int32(be_port, "9060");
 // port for brpc
 DEFINE_Int32(brpc_port, "8060");
 
+// port for brpc
+DEFINE_Int32(arrow_flight_port, "8070");
```
Comment (yiguolei) [inline be/src/common/config.cpp:64]: add the config to conf/be.conf then user will know there is a port he should config.

File: be/src/runtime/result_buffer_mgr.h [hunk 2]
```diff
@@ -53,6 +54,10 @@ class ResultBufferMgr {
                          int exec_timeout);
 
     void fetch_data(const PUniqueId& finst_id, GetResultBatchCtx* ctx);
+    Status fetch_data(const TUniqueId& finst_id, std::shared_ptr<arrow::RecordBatch>* result);
```
Comment (yiguolei) [inline be/src/runtime/result_buffer_mgr.h:57]: Do we need modify this file???

File: be/src/service/arrow_flight/flight_sql_statement_batch_reader.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+std::shared_ptr<arrow::Schema> DorisStatementBatchReader::schema() const {
+    return schema_;
+}
+
+DorisStatementBatchReader::DorisStatementBatchReader(std::shared_ptr<DorisStatement> statement,
+                                                     std::shared_ptr<arrow::Schema> schema)
+        : statement_(std::move(statement)), schema_(std::move(schema)) {}
+
+arrow::Result<std::shared_ptr<DorisStatementBatchReader>> DorisStatementBatchReader::Create(
```
Comment (yiguolei) [inline be/src/service/arrow_flight/flight_sql_statement_batch_reader.cpp:39]: create...
using doris's code format , not use arrow code format

File: be/src/common/config.cpp [hunk 4]
```diff
@@ -60,6 +60,9 @@ DEFINE_Int32(be_port, "9060");
 // port for brpc
 DEFINE_Int32(brpc_port, "8060");
 
+// port for brpc
```
Comment (yiguolei) [inline be/src/common/config.cpp:63]: comment is wrong

File: be/src/runtime/buffer_control_block.cpp [hunk 5]
```diff
@@ -119,7 +119,7 @@ Status BufferControlBlock::add_batch(std::unique_ptr<TFetchDataResult>& result)
 
     int num_rows = result->result_batch.rows.size();
 
-    while ((!_batch_queue.empty() && _buffer_rows > _buffer_limit) && !_is_cancelled) {
+    while ((!_mysql_batch_queue.empty() && _buffer_rows > _buffer_limit) && !_is_cancelled) {
```
Comment (yiguolei) [inline be/src/runtime/buffer_control_block.cpp:122]: maybe we need another pr to set buffer limit from session variable. Then for arrow flight we may need change the session variable to get better performance.

File: be/src/runtime/buffer_control_block.cpp [hunk 6]
```diff
@@ -129,18 +129,17 @@ Status BufferControlBlock::add_batch(std::unique_ptr<TFetchDataResult>& result)
...
+            std::vector<std::string>& back_rows = _mysql_batch_queue.back()->result_batch.rows;
             std::vector<std::string>& result_rows = result->result_batch.rows;
             back_rows.insert(back_rows.end(), std::make_move_iterator(result_rows.begin()),
                              std::make_move_iterator(result_rows.end()));
         } else {
-            _batch_queue.push_back(std::move(result));
+            _mysql_batch_queue.push_back(std::move(result));
         }
         _buffer_rows += num_rows;
-        _data_arrival.notify_one();
```
Comment (yiguolei) [inline be/src/runtime/buffer_control_block.cpp:143]: better not delete this line.
it seems useless but it is very dangerous to delete it.

File: be/src/service/arrow_flight/flight_sql_service.h [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+namespace doris {
+namespace flight {
+
+class FlightSqlServer : public arrow::flight::sql::FlightSqlServerBase {
+public:
+    ~FlightSqlServer() override;
+
+    static arrow::Result<std::shared_ptr<FlightSqlServer>> create();
+
+    arrow::Result<std::unique_ptr<arrow::flight::FlightDataStream>> DoGetStatement(
```
[the remark this answers is not in the document]
Comment (xinyiZzz) [reply be/src/service/arrow_flight/flight_sql_service.h:33]: `DoGetStatement` override `arrow::flight::sql::FlightSqlServerBase:: DoGetStatement` cannot be renamed = =

File: be/src/service/arrow_flight/flight_sql_statement_batch_reader.cpp (hunk 3 above)
Comment (xinyiZzz) [reply be/src/service/arrow_flight/flight_sql_statement_batch_reader.cpp]: ok, done

File: be/src/common/config.cpp (hunk 4 above)
Comment (xinyiZzz) [reply be/src/common/config.cpp:63]: fix

File: be/src/runtime/result_buffer_mgr.h (hunk 2 above)
Comment (xinyiZzz) [reply be/src/runtime/result_buffer_mgr.h:57]: done

File: be/src/runtime/buffer_control_block.cpp (hunk 5 above)
Comment (xinyiZzz) [reply be/src/runtime/buffer_control_block.cpp:122]: meaningful

**Comment by xinyiZzz** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.07 seconds
 stream load tsv:          531 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162028603 Bytes

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**